native_client/client.cc

```diff
@@ -40,7 +40,7 @@ LocalDsSTT(ModelState* aCtx, const short* aBuffer, size_t aBufferSize,
 
   clock_t ds_start_time = clock();
 
-  res.string = DS_SpeechToText(aCtx, aBuffer, aBufferSize, aSampleRate);
+  res.string = DS_SpeechToText(aCtx, aBuffer, aBufferSize, aSampleRate, extended_metadata);
```

Comment:
Here’s the problem: the API and client are pretty well insulated from each other and there isn’t really a way to pass user options from the client directly to the API.

I made some tweaks so you can now set flags on the ModelState object. This means that no existing API calls need to be changed. Would this work?

Reply:
@dabinat I'm really not convinced so far. I don't understand why you would need specific flags: is this timing computation very intensive? You have not answered that. From looking at the code, not much.

If it's not intensive, keep the flow simple, compute it and store it, and augment the API to make it accessible.

Reply:
I’m not sure it’s necessary to store it. My code is doing exactly the same as before, except instead of immediately converting the Output struct to a string, it does extra processing on it first. So if it was not necessary to store those string characters anywhere in the previous version, it shouldn’t be necessary to store the timing info either, as both make exactly the same call to the CTC decoder.

I’m happy to make whatever changes you’d like, but the way you described it is not how the API functioned prior to this patch (unless I’m misunderstanding you).

So we could simplify it by passing everything through the metadata function and then output it differently depending on what the user requests. It did seem a little inefficient to parse out the words and then string them back together to me, but from a computational point of view it would make no difference.

Also, the output doesn’t necessarily have to be JSON - I only added it because it was requested. It could very well be a textual output instead, or it could output the words and timings separately.

Reply:
> I’m happy to make whatever changes you’d like, but the way you described it is not how the API functioned prior to this patch (unless I’m misunderstanding you).

I'm not sure what you are talking about. I'm precisely asking to keep the behavior as close as possible as what we have right now. Adding extra parameters, changing the behavior of the output is not really doing that. If we don't have to store, then it's perfect: we just add a `StreamingState::finishStreamXXX` that does the extra job and then we keep the current API untouched, and we get access to the extra infos?

Reply:
> Also, the output doesn’t necessarily have to be JSON - I only added it because it was requested. It could very well be a textual output instead, or it could output the words and timings separately.

`JSON` is fine by me. The only alternative that **might** make sense would be exposing a lower-level data structure, but then it's more painful to write bindings. This kind of output is, IMHO, supposed to be readable by a machine, not a human.

Reply:
@dabinat Does it seems okay to you? Or is there anything that is unclear?

Reply:
@dabinat So, basically, my proposal is: do not expose the feature via `DS_SpeechToText` but rather with some lower-level call, that would be parallel to `StreamingState::finishStream()`.